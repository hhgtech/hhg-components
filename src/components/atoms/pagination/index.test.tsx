import React from 'react'

import { render, cleanup } from '@testing-library/react'
import 'jest-styled-components'

import { Pagination } from '.'
import MockTheme from '../../../misc/mockTheme'

afterEach(cleanup)

test('it renders', () => {
  const rendered = render(
    <MockTheme>
      <Pagination
        pagesAmount={10}
        currentPage={3}
        onChange={(page) => {
          console.log(page)
        }}
      />
    </MockTheme>,
  )

  const self = rendered.container.firstChild
  expect(self).toMatchSnapshot()
})
