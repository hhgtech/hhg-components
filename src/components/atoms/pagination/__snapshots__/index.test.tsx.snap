// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 [data-page-active] {
  background: #2d87f3;
  color: #fff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 32px;
  height: 32px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
  color: #262626;
  cursor: pointer;
  font-size: 14px;
  font-weight: 700;
  box-sizing: border-box;
  border: 1px solid #d9d9d9;
}

.c1 > svg {
  margin: auto;
  width: 10px;
  height: 10px;
  stroke: #2d87f3;
}

.c1[data-last-page] {
  opacity: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 32px;
  height: 32px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
  color: #262626;
  cursor: pointer;
  font-size: 14px;
  font-weight: 700;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      fill="none"
      height="10"
      viewBox="0 0 6 10"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5 9L1 5L5 1"
      />
    </svg>
  </div>
  <li
    class="c2"
  >
    1
  </li>
  <li
    class="c2"
  >
    2
  </li>
  <li
    class="c2"
    data-page-active="true"
  >
    3
  </li>
  <li
    class="c2"
  >
    4
  </li>
  <li
    class="c2"
  >
    5
  </li>
  <div
    class="c1"
  >
    <svg
      fill="none"
      height="10"
      viewBox="0 0 6 10"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 1L5 5L1 9"
      />
    </svg>
  </div>
</div>
`;
